scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroupInternals.scala

```diff
@@ -27,11 +26,35 @@ import scodec.bits.BitVector
 import scala.concurrent.Promise
 import scala.util.control.NonFatal
 import io.netty.channel.EventLoop
+import io.netty.handler.codec.{LengthFieldBasedFrameDecoder, LengthFieldPrepender, TooLongFrameException}
 import io.netty.handler.proxy.Socks5ProxyHandler
+import scodec.Attempt.{Failure, Successful}
+import scodec.Codec
 
 private[dynamictls] object DynamicTLSPeerGroupInternals {
+  def buildFramingCodecs(config: FramingConfig): (LengthFieldBasedFrameDecoder, LengthFieldPrepender) = {
```

Comment:
If you say the Netty constructors do checks and raise exceptions, maybe they should be constructed in a `Try`?

Or have some property based tests to see that no combination of parameters can result in a config that passes your checks and then fails netty.

Reply:
as I said  in some comment, if constructed from properly created `FramingConfig` they will be no exceptions i.e `FramingConfig` does the same checks as netty codecs.

Reply:
Okay fair enough, then there's no point in property based testing. The fact that encodings can raise exceptions even with configuration that this accepts is more worrying.